source/common/http/header_utility.h

```diff
@@ -215,6 +215,11 @@ class HeaderUtility {
    */
   static void stripTrailingHostDot(RequestHeaderMap& headers);
 
+  /**
+   * @return bool true if the provided host has a port, false otherwise.
+   */
+  static bool hostHasPort(absl::string_view host);
```

Comment:
nit: Should this perhaps be authority instead of host? (Maybe not since we already have stripPortFromHost?)

Reply:
yeah I was going for consistency with below